gpkitmodels/GP/aircraft/tail/tail_boom.py

```diff
@@ -153,74 +118,64 @@ class HorizontalBoomBending(Model):
 
-        I0 = self.I0 = tailboom.I0
+        Beam.qbarFun = [1e-10]*N
+        Beam.SbarFun = [1.]*N
+        beam = Beam(N)
+
+        I = self.I = tailboom.I
         l = self.l = tailboom.l
         S = self.S = htail.planform.S
-        E = self.E = tailboom.E
-        k = self.k = tailboom.k
+        E = self.E = tailboom.material.E
+        Sy = self.Sy = tailboom.Sy
         rhosl = self.rhosl = state.rhosl
         Vne = self.Vne = state.Vne
         CLmax = htail.planform.CLmax
-
-        return [F >= 0.5*rhosl*Vne**2*S*CLmax,
-                th >= F*l**2/E/I0*(1+k)/2,
-                th <= thmax,
-               ]
-
-class TailBoom(Model):
-    """ Tail Boom Model
-
-    Variables
-    ---------
-    l                           [ft]        tail boom length
-    E           150e9           [N/m^2]     Youngs modulus for carbon fiber
-    k           0.8             [-]         tail boom taper index
-    kfac        self.minusk2    [-]         (1-k/2)
-    I0                          [m^4]       tail boom moment of inertia
-    d0                          [in]        tail boom diameter
-    t0                          [in]        tail boom thickness
-    tmin        0.25            [mm]        minimum tail boom thickness
-    rhocfrp     1.6             [g/cm^3]    density of CFRP
-    W                           [lbf]       tail boom weight
-    J                           [m^4]       tail boom polar moment of inertia
-    S                           [ft^2]      tail boom surface area
-    mfac        1.0             [-]         tail boom margin factor
-
-    Upper Unbounded
-    ---------------
-    W
-
-    Lower Unbounded
-    ---------------
-    S, J, l, I0
-
-    LaTex Strings
-    -------------
-    kfac        (1-k/2)
-    I0          I_0
-    d0          d_0
-    t0          t_0
-    tmin        t_{\\mathrm{min}}
-    rhocfrp     \\rho_{\\mathrm{CFRP}}
-    mfac        m_{\\mathrm{fac}}
-
-    """
-
-    minusk2 = lambda self, c: 1-c[self.k]/2.
-    flight_model = TailBoomAero
-    hbending = HorizontalBoomBending
-    vbending = VerticalBoomBending
-    vtorsion = VerticalBoomTorsion
-
-    def setup(self):
-        exec parse_variables(TailBoom.__doc__)
-
-        return [I0 <= np.pi*t0*d0**3/8.0,
-                W/mfac >= np.pi*g*rhocfrp*d0*l*t0*kfac,
-                t0 >= tmin,
-                J <= np.pi/8.0*d0**3*t0,
-                S == l*np.pi*d0,
-               ]
-
+        deta = tailboom.deta
+        sigma = tailboom.material.sigma
+
+        return beam, [beam["dx"] == deta,
+                      F >= 0.5*rhosl*Vne**2*S,
+                      beam["\\bar{EI}"] <= E*I/F/l**2/2,
+                      Mr >= beam["\\bar{M}"][:-1]*F*l,
+                      sigma >= Mr/Sy,
+                      th == beam["\\theta"][-1],
+                      beam["\\bar{\\delta}"][-1]*CLmax <= kappa
+                     ]
+
+def makeTailBoom(N=2, tailboomSpar=TubeSpar):
```

Comment:
I wonder if there's a way to do this with classes instead of factories (the fancy name for functions-that-make-classes)...will take a look

Reply:
That would be great if you could find a better way to do this.